src/pow.cpp

```diff
@@ -71,6 +71,54 @@ unsigned int CalculateNextWorkRequired(const CBlockIndex* pindexLast, int64_t nF
     return bnNew.GetCompact();
 }
 
+// Check that on difficulty adjustments, the new difficulty does not increase
+// or decrease beyond the permitted limits.
+bool PermittedDifficultyTransition(const Consensus::Params& params, int64_t height, uint32_t old_nbits, uint32_t new_nbits)
+{
+    if (params.fPowAllowMinDifficultyBlocks) return true;
```

Comment:
splitting hairs maybe but should this also check new_bits is at/above powLimit?

Reply:
For the purposes of what is needed for the security analysis of the overall header commitment scheme, the only requirements are that this verifies (a) that the difficulty doesn't change on non-2016-multiple blocks and (b) doesn't go up or down too quickly. Its goal is forcing the attacker to spread out their attempted PoW over many blocks, rather than just one (because creating N blocks with each W work is much harder than creating one block with NW work, if the hashpower available to the attacker is less than NW).

No objection to checking whatever can be checked with the provided arguments, but I think the current code just chooses to check exactly what is needed, and document it, rather than verify everything possible.

Reply:
I think I prefer to just not check anything for chains where proof-of-work isn't an anti-DoS mechanism for us; seems like it avoids a false sense of security (and it doesn't accomplish anything anyway, as an attacker would just avoid triggering the check).